backend/src/services/trading_service.py

```
@@ -115,57 +137,64 @@ async def execute_buy_order(self, token_address: str, amount_sol: float, slippag
                 "timestamp": datetime.now().isoformat()
             }
             if self.socketio:
                 self.socketio.emit('trade_executed', trade_data)
-            return {"success": True, "message": f"Successfully bought {amount_sol} SOL worth of {token_address}", "transaction_id": transaction_id, "token_address": token_address, "amount_sol": amount_sol, "slippage": slippage, "status": "confirmed"}
+
+            # Record trade in DB
+            await asyncio.to_thread(record_trade, trade_data)
+
+            return {"success": True, "message": f"Successfully bought {amount_sol} SOL worth of {token_address}", "transaction_id": transaction_id, "token_address": token_address, "amount_sol": amount_sol, "slippage": slippage, "status": "confirmed" if confirmed else "pending", "price_usd": current_price}
 
         except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)